control/velocity_controller_lqr/scripts/velocity_controller_lqr_node.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+import numpy as np
+import rclpy
+from geometry_msgs.msg import Wrench
+from nav_msgs.msg import Odometry
+from rclpy.node import Node
+from velocity_controller_lqr.velocity_controller_lqr_lib import LQRController
+from vortex_msgs.msg import LOSGuidance
+
+
+class LinearQuadraticRegulator(Node):
+    def __init__(self):
+        super().__init__("velocity_controller_lqr_node")
+
+        self.declare_parameter("odom_path", "/nucleus/odom")
+        self.declare_parameter("guidance_path", "/guidance/los")
+        self.declare_parameter("thrust_path", "/thrust/wrench_input")
+
+        odom_path = self.get_parameter("odom_path").value
+        guidance_path = self.get_parameter("guidance_path").value
+        thrust_path = self.get_parameter("thrust_path").value
+
+        # SUBSRCIBERS -----------------------------------
+        self.nucleus_subscriber = self.create_subscription(
+            Odometry, odom_path, self.nucleus_callback, 20
+        )
+        self.guidance_subscriber = self.create_subscription(
+            LOSGuidance, guidance_path, self.guidance_callback, 20
+        )
+
+        # PUBLISHERS ------------------------------------
+        self.publisherLQR = self.create_publisher(Wrench, thrust_path, 10)
+
+        # TIMERS ----------------------------------------
+        self.control_timer = self.create_timer(0.1, self.controller)
+
+        # ROS2 SHENNANIGANS with parameters
+        self.declare_parameter("max_force", 99.5)
+        max_force = self.get_parameter("max_force").value
+
+        self.declare_parameter("q_surge", 75)
+        self.declare_parameter("q_pitch", 175)
+        self.declare_parameter("q_yaw", 175)
+
+        self.declare_parameter("r_surge", 0.3)
+        self.declare_parameter("r_pitch", 0.4)
+        self.declare_parameter("r_yaw", 0.4)
+
+        self.declare_parameter("i_surge", 0.3)
+        self.declare_parameter("i_pitch", 0.4)
+        self.declare_parameter("i_yaw", 0.3)
+
+        q_surge = self.get_parameter("q_surge").value
+        q_pitch = self.get_parameter("q_pitch").value
+        q_yaw = self.get_parameter("q_yaw").value
+
+        r_surge = self.get_parameter("r_surge").value
+        r_pitch = self.get_parameter("r_pitch").value
+        r_yaw = self.get_parameter("r_yaw").value
+
+        i_surge = self.get_parameter("i_surge").value
+        i_pitch = self.get_parameter("i_pitch").value
+        i_yaw = self.get_parameter("i_yaw").value
```

Comment:
would be nice to have all the declarations and getting in a separate method, so that the init is not so large